server/types/base_ids.go

```diff
@@ -78,63 +78,92 @@ const (
 	DoltgresTypeBaseID_Xid         = DoltgresTypeBaseID(SerializationID_Xid)
 )
 
+// TypeAlignment represents the alignment required when storing a value of this type.
+type TypeAlignment string
+
+const (
+	TypeAlignment_Char   TypeAlignment = "c"
+	TypeAlignment_Short  TypeAlignment = "s"
+	TypeAlignment_Int    TypeAlignment = "i"
+	TypeAlignment_Double TypeAlignment = "d"
+)
+
 // TypeCategory represents the type category that a type belongs to. These are used by Postgres to group similar types
 // for parameter resolution, operator resolution, etc.
-type TypeCategory uint8
+type TypeCategory string
 
 const (
-	TypeCategory_Unknown TypeCategory = iota
-	TypeCategory_ArrayTypes
-	TypeCategory_BooleanTypes
-	TypeCategory_CompositeTypes
-	TypeCategory_DateTimeTypes
-	TypeCategory_EnumTypes
-	TypeCategory_GeometricTypes
-	TypeCategory_NetworkAddressTypes
-	TypeCategory_NumericTypes
-	TypeCategory_PseudoTypes
-	TypeCategory_RangeTypes
-	TypeCategory_StringTypes
-	TypeCategory_TimespanTypes
-	TypeCategory_UserDefinedTypes
-	TypeCategory_BitStringTypes
-	TypeCategory_XMLTypes
+	TypeCategory_ArrayTypes          TypeCategory = "A"
+	TypeCategory_BooleanTypes        TypeCategory = "B"
+	TypeCategory_CompositeTypes      TypeCategory = "C"
+	TypeCategory_DateTimeTypes       TypeCategory = "D"
+	TypeCategory_EnumTypes           TypeCategory = "E"
+	TypeCategory_GeometricTypes      TypeCategory = "G"
+	TypeCategory_NetworkAddressTypes TypeCategory = "I"
+	TypeCategory_NumericTypes        TypeCategory = "N"
+	TypeCategory_PseudoTypes         TypeCategory = "P"
+	TypeCategory_RangeTypes          TypeCategory = "R"
+	TypeCategory_StringTypes         TypeCategory = "S"
+	TypeCategory_TimespanTypes       TypeCategory = "T"
+	TypeCategory_UserDefinedTypes    TypeCategory = "U"
+	TypeCategory_BitStringTypes      TypeCategory = "V"
+	TypeCategory_UnknownTypes        TypeCategory = "X"
+	TypeCategory_InternalUseTypes    TypeCategory = "Z"
 )
 
 // baseIDArrayTypes contains a map of all base IDs that represent array variants.
 var baseIDArrayTypes = map[DoltgresTypeBaseID]DoltgresArrayType{}
 
+// baseIDValidTypes contains a map of all base IDs that are valid Postgres type displayed in pg_catalog tables.
+var baseIDValidTypes = map[string]DoltgresValidType{
+	internalChar.BaseName(): internalChar, // add internal type
+}
```

Comment:
Since this is only used for `pg_type`, we can remove this map and use the `typesFromBaseID` map. Within `pg_type`, we can filter all of the array types out (in addition to `null`), leaving only the base types, and we can explicitly reference the `InternalChar` type as well since we only need it for the output there.